ui/app/services/auth.js

```diff
@@ -216,9 +218,17 @@ export default Service.extend({
   },
 
   calculateExpiration(resp) {
-    const now = this.now();
+    let tokenExpirationEpoch;
     const ttl = resp.ttl || resp.lease_duration;
-    const tokenExpirationEpoch = now + ttl * 1e3;
+    const now = this.now();
+
+    if (resp.type === 'batch') {
```

Comment:
since this is only for token mount of type=batch, I wonder if we should instead check for the presence of `resp.expire_time` here and return this static expiration if the token is not renewable. We will have to double check that all non-renewable token responses include `expire_time` if we do take that route

Reply:
that's a good point, especially since the next line assumes that `expiry_time` exists. i updated the code to reflect this.